test/single-spa-layout.test-d.ts

```diff
@@ -1,12 +1,26 @@
 import { expectError, expectType } from "tsd";
-import { constructRoutes } from "../src/single-spa-layout";
-
-expectType<any>(
-  constructRoutes({
-    routes: [],
-  })
-);
+import { constructRoutes, matchRoute } from "../src/single-spa-layout";
 
+// test constructRoutes
 expectError(constructRoutes());
 
 expectError(constructRoutes({}));
+
+const routes = constructRoutes({
+  routes: [
+    {
+      type: "route",
+      path: "/app1",
+      routes: [{ type: "application", name: "app1" }],
+    },
+  ],
+});
+
+// test matchRoute
+const matchedRoutes = matchRoute(routes, "/");
```

Comment:
This file is actually one of the easier files to get a brief overview of what's happening in this PR